File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/DomainOrder.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.model.DomainOrderServices;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Ordering metadata for domain registration and transfers.
 */
@Schema(description = "Ordering metadata for domain registration and transfers.")
@Validated
@NotUndefined



public class DomainOrder   {
  @JsonProperty("whoisPrivacyCost")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String whoisPrivacyCost = null;

  @JsonProperty("services")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private DomainOrderServices services = null;

  @JsonProperty("tldServices")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private Object tldServices = null;


  public DomainOrder whoisPrivacyCost(String whoisPrivacyCost) { 

    this.whoisPrivacyCost = whoisPrivacyCost;
    return this;
  }

  /**
   * Cost of Whois Privacy
   * @return whoisPrivacyCost
   **/
  
  @Schema(example = "5", description = "Cost of Whois Privacy")
  
  public String getWhoisPrivacyCost() {  
    return whoisPrivacyCost;
  }



  public void setWhoisPrivacyCost(String whoisPrivacyCost) { 
    this.whoisPrivacyCost = whoisPrivacyCost;
  }

  public DomainOrder services(DomainOrderServices services) { 

    this.services = services;
    return this;
  }

  /**
   * Get services
   * @return services
   **/
  
  @Schema(description = "")
  
@Valid
  public DomainOrderServices getServices() {  
    return services;
  }



  public void setServices(DomainOrderServices services) { 
    this.services = services;
  }

  public DomainOrder tldServices(Object tldServices) { 

    this.tldServices = tldServices;
    return this;
  }

  /**
   * Map of TLDs to domain service IDs.
   * @return tldServices
   **/
  
  @Schema(description = "Map of TLDs to domain service IDs.")
  
  public Object getTldServices() {  
    return tldServices;
  }



  public void setTldServices(Object tldServices) { 
    this.tldServices = tldServices;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DomainOrder domainOrder = (DomainOrder) o;
    return Objects.equals(this.whoisPrivacyCost, domainOrder.whoisPrivacyCost) &&
        Objects.equals(this.services, domainOrder.services) &&
        Objects.equals(this.tldServices, domainOrder.tldServices);
  }

  @Override
  public int hashCode() {
    return Objects.hash(whoisPrivacyCost, services, tldServices);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DomainOrder {\n");
    
    sb.append("    whoisPrivacyCost: ").append(toIndentedString(whoisPrivacyCost)).append("\n");
    sb.append("    services: ").append(toIndentedString(services)).append("\n");
    sb.append("    tldServices: ").append(toIndentedString(tldServices)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/DomainOrder.cpp


#include "DomainOrder.h"

using namespace Tiny;

DomainOrder::DomainOrder()
{
	whoisPrivacyCost = std::string();
	services = DomainOrder_services();
	tldServices = null;
}

DomainOrder::DomainOrder(std::string jsonString)
{
	this->fromJson(jsonString);
}

DomainOrder::~DomainOrder()
{

}

void
DomainOrder::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *whoisPrivacyCostKey = "whoisPrivacyCost";

    if(object.has_key(whoisPrivacyCostKey))
    {
        bourne::json value = object[whoisPrivacyCostKey];



        jsonToValue(&whoisPrivacyCost, value, "std::string");


    }

    const char *servicesKey = "services";

    if(object.has_key(servicesKey))
    {
        bourne::json value = object[servicesKey];




        DomainOrder_services* obj = &services;
		obj->fromJson(value.dump());

    }

    const char *tldServicesKey = "tldServices";

    if(object.has_key(tldServicesKey))
    {
        bourne::json value = object[tldServicesKey];




        Object* obj = &tldServices;
		obj->fromJson(value.dump());

    }


}

bourne::json
DomainOrder::toJson()
{
    bourne::json object = bourne::json::object();





    object["whoisPrivacyCost"] = getWhoisPrivacyCost();







	object["services"] = getServices().toJson();






	object["tldServices"] = getTldServices().toJson();


    return object;

}

std::string
DomainOrder::getWhoisPrivacyCost()
{
	return whoisPrivacyCost;
}

void
DomainOrder::setWhoisPrivacyCost(std::string whoisPrivacyCost)
{
	this->whoisPrivacyCost = whoisPrivacyCost;
}

DomainOrder_services
DomainOrder::getServices()
{
	return services;
}

void
DomainOrder::setServices(DomainOrder_services services)
{
	this->services = services;
}

Object
DomainOrder::getTldServices()
{
	return tldServices;
}

void
DomainOrder::setTldServices(Object tldServices)
{
	this->tldServices = tldServices;
}



